.gitignore

```diff
@@ -2,4 +2,6 @@
 .cache
 .vagrant
 *.retry
-test.sh
+.idea/*
+demo/client-output/*
+*.iml
```

Comment:
@orenbm i'd suggest newline at EOF. there's a couple other files in the PR without. (https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)